shell/edit/catalog.cattle.io.clusterrepo.vue

```diff
@@ -39,6 +40,21 @@ export default {
       return this.$store.getters['cluster/all'](NAMESPACE)[0]?.id;
     }
   },
+
+  methods: {
+    onTargetChange(isGit) {
+      // reset entered value when switching options
+      if (isGit) {
+        set(this.value, 'spec.url', '');
```

Comment:
you have `Vue.set` also. Not saying this needs a change but `Vue.set` is the pattern i am more used to use.

Reply:
@jordojordo you're absolutely right, child components should only trigger events to the parent component in order to change their prop values, but unfortunately `this.value` has been treated widely in the app with the anti-pattern way. Here in this case, it requires changes in `shell/components/ResourceDetail/index.vue` that is a generic component and needs lots of testing and time to make sure no side effect happens.